core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistry.java

```diff
@@ -443,6 +442,105 @@ protected boolean matches(
     }
   }
 
+  @Nullable
+  protected DataType inferCqlTypeFromValue(@NonNull Object value) {
+    if (value instanceof List) {
+      List<?> list = (List<?>) value;
+      if (list.isEmpty()) {
+        return CQL_TYPE_FOR_EMPTY_LISTS;
```

Comment:
why `CQL_TYPE_FOR_EMPTY_LISTS` carries `Boolean` as a type? is it safe? (same question for `Set`, `Map`)

Reply:
It's explained [here](https://github.com/datastax/java-driver/pull/1377/files#diff-34d4c85a5bfd4f5ccb02c01deb989c99R682-R684): an empty collection is serialized in the same way for all element types.